pvlib/tests/iotools/test_crn.py

```diff
@@ -39,7 +49,7 @@ def testfile_problems():
     return DATA_DIR / 'CRN_with_problems.txt'
```

Comment:
To fully test the new mapping of values to nans, we could modify this file and the corresponding test results. I believe I created the test file by extracting a section of a real file because of the funky character that shows up. But seems fine to make further changes. For example, change `   190` to `-99999` in the first line.

Reply:
I had to change a '3' to 3.0 in the `test_read_crn_problems` - is this ok @wholmgren?

Reply:
Strange. The module's `DTYPES` specifies that should be `str`. This is from the [documentation](https://www.ncei.noaa.gov/pub/data/uscrn/products/subhourly01/)

>    6    CRX_VN  [6 chars]  cols 35 -- 40 
>           The version number of the station datalogger program that was in 
>           effect at the time of the observation. Note: This field should be 
>           treated as text (i.e. string).

So I'm inclined to try to avoid this change.

I'm not sure why this change might have been needed to make a test pass... Does `replace` mangle dtypes in a way that `where` does not?

Reply:
I can't replicate it any longer... it seems to work fine now so I have changed it back to be a string. Apologies.